### PR TITLE
Check "Not before" date

### PR DESCRIPTION
**Pull request by navzam**
Fixes issue #23 

Main changes:

- Added check for `Not Before` date in certificate
- Modified existing `Not After` test to adopt new error message

I haven't added a test for this `Not Before` check yet. It looks like the `Not After` test uses an old certificate, which makes sense. Unfortunately we can't use a future certificate for a `Not Before` test :) I guess we have to build out a fake certificate. Ideas welcome.

### CODE REVIEW

File: validate-cert.js [hunk 1]
```diff
@@ -12,12 +12,19 @@ module.exports = function validate(pem_cert) {
...
+    var now = new Date().getTime()
     var notAfter = new Date(cert.validity.notAfter)
-    var remainingDays = notAfter.getTime() - new Date().getTime()
-    // check that the signing certificate has not expired (examine both the Not
-    // Before and Not After dates)
-    if (remainingDays < 1) {
-      return 'certificate expiration check failed'
+    var timeUntilAfter = notAfter.getTime() - now
+    if (timeUntilAfter < 1) {
+      return 'certificate Not After check failed'
```
Comment (mreinstein) [inline validate-cert.js]: this logic is kind of hard to read...I'm not sure if this check is validating the end date or start date of the given certificate. can you simplify/rename what is happening here so that's more clear?

Comment (mreinstein) [inline validate-cert.js]: can you make the error string more understandable to a human? an example of something that might be more readable: 

`invalid certificate date (expired)`

File: validate-cert.js [hunk 2]
```diff
@@ -12,12 +12,19 @@ module.exports = function validate(pem_cert) {
...
+    }
+
+    // check that the signing certificate has not expired (examine Not Before)
+    var notBefore = new Date(cert.validity.notBefore)
+    var timeSinceBefore = now - notBefore.getTime()
+    if (timeSinceBefore < 1) {
+      return 'certificate Not Before check failed'
```
Comment (mreinstein) [inline validate-cert.js]: similar to above comments, kind of hard to follow this logic. I suspect better variable names might help.

also as above, a more human readable error string. an example of something that might be more readable: 
`invalid certificate date (start date is in the future)`

**Comment by mreinstein** [comment]
@navzam looks like a good start! I've added some feedback on very minor changes.

File: validate-cert.js (hunk 1 above)
Comment (navzam) [inline validate-cert.js]: Sure. The previous message was "certificate expiration check failed" so I was staying along those lines, but I can definitely make it human readable

Comment (mreinstein) [inline validate-cert.js]: > previous message was "certificate expiration check failed" so I was staying 

Yeah, it's my bad. I wrote it originally and it's not super high quality. Just thinking while you're in here changing those lines, would be awesome to make it more readable etc.

**Comment by dblock** [comment]
Maybe stubbing `new Date()` would allow a test for this.

**Comment by navzam** [comment]
Maybe we could create an invalid cert with a future `Not Before` date. Similar to how [this test](https://github.com/mreinstein/alexa-verifier/blob/master/test/validate-cert.js#L91-L95) creates one.
